reVX/plexos/base.py

```diff
@@ -561,3 +586,140 @@ def _ammend_output_meta(self, row_idx, sc_gids, res_gids, gen_gids,
             self._output_meta.at[index, 'res_gids'] += res_gids
             self._output_meta.at[index, 'gen_gids'] += gen_gids
             self._output_meta.at[index, 'res_built'] += res_built
+
+    @staticmethod
+    def tz_convert_profiles(profiles, timezone):
+        """Convert profiles to local time and forward/back fill missing data.
+
+        Parameters
+        ----------
+        profiles : np.ndarray
+            Profiles of shape (time, n_plants) in UTC
+        timezone : str
+            Timezone for output generation profiles. This is a string that will
+            be passed to pytz.timezone() e.g. US/Pacific, US/Mountain,
+            US/Central, US/Eastern, or UTC. For a list of all available
+            timezones, see pytz.all_timezones
+
+        Returns
+        -------
+        profiles : np.ndarray
+            Profiles of shape (time, n_plants) in timezone
+        """
+
+        logger.info('Converting profiles timezone to {}'.format(timezone))
+
+        if len(profiles) < 8760:
+            msg = ('Cannot use profiles that are not at least hourly! '
+                   'Received shape {}'.format(profiles.shape))
+            logger.error(msg)
+            raise RuntimeError(msg)
+
+        steps_per_hour = len(profiles) // 8760
+
+        # use jan 1 to avoid daylight savings
+        date = datetime.datetime(2011, 1, 1)
+        date = pytz.timezone(timezone).localize(date)

```

Comment:
Whoa this is cool, I have never heard of `pytz`. Looks super useful for these annoying timezone conversions. And actually, now that I think of it, maybe I can use it to replace that ugly utc offset dict in the `Cities` class in `ntps_load`. Will explore this!

Reply:
Yeah it's still painful but helps a little bit haha